lib/function/probability/factorial.js

```diff
@@ -55,31 +55,36 @@ module.exports = function (math) {
 
       n = n.toNumber();   // Should definitely be below Number.MAX_VALUE  
       if (n < smallBigFacs.length) {
-        return BigNumber.convert(smallBigFacs[n]).toSD(math.precision);
+        return BigNumber.convert(smallBigFacs[n]).toSD(config.precision);
       }
 
+      // Be wary of round-off errors
+      var precision = config.precision + 4;
```

Comment:
I think I will go conservative here and do `floor(log(n))` as most cases won't notice the difference, and larger cases won't be able to check against wolfram, so it is important that they are correct more so than fast.